Applications/spire/source/charting/charting_technicals_panel.cpp

```diff
@@ -111,26 +91,50 @@ ChartingTechnicalsPanel::ChartingTechnicalsPanel(TechnicalsModel& model)
   m_volume_text_label->setStyleSheet(text_label_stylesheet);
   ohlc_layout->addWidget(m_volume_text_label);
   ohlc_layout->addSpacing(scale_width(3));
-  m_volume_value_label = new QLabel(m_item_delegate->displayText(
-    QVariant::fromValue(m_model->get_volume()), QLocale()), this);
+  m_volume_value_label = new QLabel("0", this);
   m_volume_value_label->setFixedHeight(scale_height(11));
   m_volume_value_label->setStyleSheet(value_label_stylesheet);
   ohlc_layout->addWidget(m_volume_value_label);
   ohlc_layout->addSpacing(scale_width(10));
   ohlc_layout->addStretch(1);
   layout->addStretch(8);
-  m_last_price_connection = m_model->connect_last_price_slot(
-    [=] (const Money& last) { on_last_price_signal(last); });
-  m_open_connection = m_model->connect_open_slot([=] (const Money& open) {
-    on_open_signal(open); });
-  m_close_connection = m_model->connect_close_slot([=] (const Money& close) {
-    on_close_signal(close); });
-  m_high_connection = m_model->connect_high_slot([=] (const Money& high) {
-    on_high_signal(high); });
-  m_low_connection = m_model->connect_low_slot([=] (const Money& low) {
-    on_low_signal(low); });
-  m_volume_connection = m_model->connect_volume_slot(
-    [=] (const Quantity& volume) { on_volume_signal(volume); });
+  m_model_promise = m_model->load();
+  m_model_promise.then([&] (auto result) {
```

Comment:
Stop making the mistake of using a ```&``` when you should use a ```=```, this can cause a crash in the best case scenario, and data corruption in the worst case scenario. Every single time you write a lambda, carefully consider how it's being used.